pkg/frontend/asyncoperationresult_get.go

```diff
@@ -64,7 +65,7 @@ func (f *frontend) _getAsyncOperationResult(ctx context.Context, r *http.Request
 
 	asyncdoc.OpenShiftCluster.Properties.ClusterProfile.PullSecret = ""
 
-	if asyncdoc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil {
+	if !iswimi.IsWimi(asyncdoc.OpenShiftCluster.Properties) {
```

Comment:
Let's remove this - it's not explicitly checking if it's a wi/mi cluster or not, just if the service principal is nil or not.